packages/asset/contracts/AssetCreate.sol

```diff
@@ -354,5 +355,194 @@ contract AssetCreate is
+            }
+        }
+
+        catalystContract.burnBatchFrom(_msgSender(), tiersToBurn, amounts);
+        vault.distribute(tiers, amounts, creators);
+        assetContract.mintBatch(_msgSender(), tokenIds, amounts, metadataHashes);
+        emit AssetBatchLazyMinted(_msgSender(), creators, tokenIds, tiers, amounts, metadataHashes);
+    }
+
+    /// @notice Creates a hash of the lazy mint data
+    /// @param creator The address of the creator
+    /// @param nonce The nonce of the creator
+    /// @param tier The tier of the asset
+    /// @param amount The amount of copies to mint
+    /// @param metadataHash The metadata hash of the asset
+    /// @param maxSupply The max supply of the asset
+    function _hashLazyMint(
+        address creator,
+        uint16 nonce,
+        uint8 tier,
+        uint256 amount,
+        string memory metadataHash,
+        uint256 maxSupply
+    ) internal view returns (bytes32 digest) {
+        digest = _hashTypedDataV4(
+            keccak256(
+                abi.encode(
+                    LAZY_MINT_TYPEHASH,
+                    creator,
+                    nonce,
+                    tier,
+                    amount,
+                    keccak256((abi.encodePacked(metadataHash))),
+                    maxSupply
+                )
+            )
+        );
+    }
+
+    /// @notice Creates a hash of the lazy mint batch data
+    /// @param creators The addresses of the creators
+    /// @param nonce The nonce of the creator
+    /// @param tiers The tiers of the assets
+    /// @param amounts The amounts of copies to mint
+    /// @param metadataHashes The metadata hashes of the assets
+    /// @param maxSupplies The max supplies of the assets
+    function _hashLazyBatchMint(
+        address[] memory creators,
+        uint16 nonce,
+        uint8[] memory tiers,
+        uint256[] memory amounts,
+        string[] memory metadataHashes,
+        uint256[] memory maxSupplies
+    ) internal view returns (bytes32 digest) {
+        digest = _hashTypedDataV4(
+            keccak256(
+                abi.encode(
+                    LAZY_MINT_BATCH_TYPEHASH,
+                    keccak256(abi.encodePacked(creators)),
+                    nonce,
+                    keccak256(abi.encodePacked(tiers)),
+                    keccak256(abi.encodePacked(amounts)),
+                    _encodeHashes(metadataHashes),
+                    keccak256(abi.encodePacked(maxSupplies))
+                )
+            )
+        );
+    }
+
+    /// @notice Set a new vault address, limited to DEFAULT_ADMIN_ROLE only
+    /// @dev Change the address of the vault that distributes the catalyst value to the creators
+    /// @param _vault The new vault
+    function setVaultAddress(address _vault) external onlyRole(DEFAULT_ADMIN_ROLE) {
+        require(_vault.isContract(), "AssetCreate: Bad vault addr");
+        vault = ILazyVault(_vault);
+    }
+
+    uint256[43] private __gap;
```

Comment:
41 ?

Reply:
I am guessing you are referring to the two constant variables that were declared? Little more context would be nice 🥲

No, those do not occupy a slot since they are constants and are baked into the bytecode, therefore the gap variable does not need to be decremented.